packages/@aws-cdk/aws-lambda/README.md

```diff
@@ -120,6 +120,62 @@ lambdaAction.outputArtifact('Out2'); // returns the named output Artifact, or th
 See [the AWS documentation](https://docs.aws.amazon.com/codepipeline/latest/userguide/actions-invoke-lambda-function.html)
 on how to write a Lambda function invoked from CodePipeline.
 
+### Lambda deployed through CodePipeline
+
+If you want to deploy your Lambda through CodePipeline,
+you need to override the Parameters that are present in the Asset of the Lambda Code.
+Note that your Lambda must be in a different Stack than your Pipeline.
+The Lambda itself will be deployed, alongside the entire Stack it belongs to,
+using a CloudFormation CodePipeline Action. Example:
+
+```typescript
+const lambdaCode = lambda.Code.asset('path/to/directory/or/zip/file'); 
+const lambda = new lambda.Function(lambdaStack, 'Lambda', {
+  code: lambdaCode,
+  handler: 'index.handler',
+  runtime: lambda.Runtime.NodeJS810,
+});
+// other resources that your Lambda needs, added to the lambdaStack...
+
+const pipeline = new codepipeline.Pipeline(pipelineStack, 'Pipeline');
+// add the source code repository containing this code to your Pipeline,
+// and the source code of the Lambda Function, if they're separate
+pipeline.addStage({
+  name: 'Source',
+  actions: [
+    // ...
+  ],
+});
+// add a build Action to your Pipeline, that calls `cdk synth` on the lambdaStack,
+// and saves it to some file, and a separate build for your Lambda source code - if needed
+pipeline.addStage({
+  name: 'Build',
+  actions: [
+    lambdaBuildAction,
+    cdkBuildAction,
+  ],
+});
+// finally, deploy your Lambda code
+const parameterOverrides: { [name: string]: any } = {};
+parameterOverrides[lambdaCode.asset.bucketNameParam.logicalId] = lambdaBuildAction.outputArtifact.bucketName;
```

Comment:
Can we not hide this? Something like:

```ts
const overrides = lambdaBuildAction.outputArtifact.becomeAsset(lambdaCode.asset);
```

Would already seem more palatable to me (better name, obviously).

Reply:
“connectToAsset”?

Reply:
I went with an `Asset#overrideWith(cpapi.Artifact)` method. So, it looks like this in code:

```ts
  parameterOverrides: {
    ...lambdaCode.asset.overrideWith(lambdaBuildAction.outputArtifact),
  },
  // ...
```

Let me know if that's acceptable.

Reply:
Can we even do that? That means the `assets` package needs to depend on the codepipeline package, no?

Reply:
Yeah, you're  right. I read better this way for me, but of course from a dependency standpoint it's weird.

How about `artifact.override(asset)`?

Reply:
Discussed offline - the consensus reached was: `artifact.overrideAsset(asset)`.

Reply:
Houston, we have a problem :/.

We can't depend on the `assets` package from `codepipeline-api`, as that causes a cycle between those 2 and the `s3` module.

Reply:
I see 2 possible options of getting out of this cycle:

1. Keep the current solution (`assets` depending on `codepipeline-api`). That's not the greatest, but `codepipeline-api` is a small package with only foundational dependencies like IAM and Events.
2. Introduce an `assets-api` package with an `IAsset` class that `Asset` implements, and depend on that from `codepipeline-api`.

Thoughts?

Reply:
Is "postpone until actions moved" an option?

I see why you might not want to do that. In that case I vote reversing the dependency arrow for now (another package is just one we're going to have to deprecate later), but with the full understanding we're going to be reversing the API before 1.0

Reply:
> Is "postpone until actions moved" an option?

Let me explore that. Since we're planning to do that anyway, it might save us some churn in the long run.

Reply:
With #2098 finally being merged, I was able to make the dependency in the correct direction (`codepipeline` package depending on `assets`). Let me know if this is good!